src/core/server/uuid/uuid_service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { manageInstanceUuid } from './manage_uuid';
+import { CoreContext } from '../core_context';
+import { Logger } from '../logging';
+import { IConfigService } from '../config';
+import { LegacyServiceDiscoverPlugins } from '../legacy/legacy_service';
+
+/**
+ * APIs to access the application's instance uuid.
+ */
+export interface UuidServiceSetup {
+  /**
+   * Retrieve the Kibana instance uuid.
+   */
+  getInstanceUuid(): string;
+}
+
+interface SetupDeps {
+  legacyPlugins: LegacyServiceDiscoverPlugins;
+}
+
+/** @internal */
+export class UuidService {
+  private readonly log: Logger;
+  private readonly configService: IConfigService;
+  private uuid: string = '';
+
+  constructor(core: CoreContext) {
+    this.log = core.logger.get('uuid');
+    this.configService = core.configService;
+  }
+
+  public async setup({ legacyPlugins }: SetupDeps) {
+    this.uuid = await manageInstanceUuid(this.configService);
+    this.log.info(`Kibana instance UUID: ${this.uuid}`);
+
+    // propagate the instance uuid to the legacy config, as it was the legacy way to access it.
+    legacyPlugins.pluginExtendedConfig.set('server.uuid', this.uuid);
```

Comment:
The legacy way to access the instance uuid was via the config... the legacy mixin was writing it during server init.

I tried to migrates the legacy usages of `config.get('server.uuid')`, however some calls are very deeply nested in places where the server, and therefor the NP API, is not accessible. So to preserve retro-compatibility, I was kinda forced to do that.

Tell me if we think this is not acceptable and if I should migrate all calls instead.

Reply:
Would it be possible to move this code into the LegacyService instead? That way we can more easily delete legacy code without having to modify this as well.

Reply:
Done. The legacy service tests mocks requirements are growing though. It was harder than expected to move the legacy config set test here.